cstore_fdw.c

```diff
@@ -347,7 +350,12 @@ CStoreProcessUtility(Node * parseTree, const char *queryString,
 	}
 	else if (nodeTag(parseTree) == T_TruncateStmt)
 	{
-		TruncateStmt *truncateStatement = (TruncateStmt *) parseTree;
+#if PG_VERSION_NUM >= 100000
+		TruncateStmt *truncateStatement =  (TruncateStmt *) parseTree;
+#else
+		/* we are modifying the parseTree, therefore we will work on a copy */
+		TruncateStmt *truncateStatement =  (TruncateStmt *) copyObject(parseTree);
```

Comment:
I think we are good for just copying parseTree here, there should not be any need for making a copy of plannedStatement.

Reply:
If we just copy the parseTree for PG >= 10, we have to modify the input PlannedStmt and restore it after the call to CALL_PREVIOUS_UTILITY.

Reply:
I did not like this approach. We will end up performing copy when any utility call is made which could be a lot.

If we can't get away with only copying parseTree, then we should revert back to your original solution of restoring table list back, and ditch all the rest.